components/kyma-environment-broker/common/gardener/client.go

```diff
@@ -25,6 +23,15 @@ func (b SecretBinding) GetSecretRefName() string {
 	return str
 }
 
+func (b SecretBinding) GetSecretRefNamespace() string {
+	str, _, err := unstructured.NestedString(b.Unstructured.Object, "secretRef", "namespace")
+	if err != nil {
+		// NOTE this is a safety net, gardener v1beta1 API would need to break the contract for this to panic
+		panic(fmt.Sprintf("SecretBinding missing field '.secretRef.namespace': %v", err))
```

Comment:
Could you elaborate why a panic needs here to take place? Why not return an error or use log.Fatal?

Reply:
returning an error would mean we will need to refactor in many places for a condition that can happen only if gardener changes their v1beta1 API to be incompatible with the current implementation, it was very briefly discussed outside of github and summarized here https://github.com/kyma-project/control-plane/pull/1527#discussion_r845069652